pkg/config/config.go

```diff
@@ -130,21 +134,47 @@ func (c *ConfigInfo) writeToFile() error {
 }
 
 // SetConfiguration modifies Odo configurations in the config file
-// as of now only being used for updatenotification
-func (c *ConfigInfo) SetConfiguration(parameter string, value bool) error {
+// as of now being used for timeout, updatenotification
+func (c *ConfigInfo) SetConfiguration(parameter string, value interface{}) error {
 	switch parameter {
+
+	case "timeout":
+		typedval, ok := value.(int)
+		if !ok {
+			return errors.Errorf("type assertion error: unable to set %s", parameter)
+		}
+		if typedval < 0 {
+			return errors.Errorf("cannot set timeout to less than 0, A value of zero means don't timeout requests")
```

Comment:
remove `, A value of zero means don't timeout requests` this isn't really needed.